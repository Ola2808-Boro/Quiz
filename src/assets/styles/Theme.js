export const theme = {
  colors: {
    backgroundColor: "#F5F7FB",
    text: "#293264",
    buttonColor: "#F5F7FB",
    circleBottom: "#DEEBF8",
    circleUp: "#FFFAD1",
    buttonBg: "#4D5B9E",
    borderAnswer: "#4D5B9E",
  },

  fontSize: {
    huge: "70px",
    xlgrand: "44px",
    lgrand: "40px",
    grand: "37px",
    xxxxl: "34px",
    xxxl: "33px",
    xxl: "30 px",
    xl: "26px",
    mediumxl: "24px",
    l: "22px",
    ml: "18px",
    m: "14px",
    s: "10px",
  },
};
