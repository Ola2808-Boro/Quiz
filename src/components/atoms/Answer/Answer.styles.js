import styled from "styled-components";
import { device } from "helpers/breakpoints";
export const StyledAnswer = styled.div`
  ${(props) => {
    if (props.isClicked) {
      return `
      
      border: 1px solid ${({ theme }) => theme.colors.borderAnswer};
   
    `;
    } else {
      return `
      border: 1px solid #d6dbf5;
   
    `;
    }
  }}
  background-color: ${(props) => (props.isClicked ? "#d6dbf5" : "transparent")};
  padding: 0.3rem 0.5rem;
  color: ${({ theme }) => theme.colors.text};
  font-size: ${({ theme }) => theme.fontSize.s};
  margin-right: 0.5rem;
  border-radius: 10px;
  text-align: center;

  @media ${device.mobileS} {
    margin: 0.25rem;
    text-align: center;
    font-size: ${({ theme }) => theme.fontSize.m};
  }
  @media ${device.mobileM} {
    font-size: ${({ theme }) => theme.fontSize.ml};
  }
  @media ${device.mobileL} {
    font-size: ${({ theme }) => theme.fontSize.ml};
  }
  @media ${device.tablet} {
    font-size: ${({ theme }) => theme.fontSize.mediumxl};
  }
  @media ${device.laptop} {
    margin: 1rem 1rem 0.5rem 0rem;
    padding: 0.5rem 0.7rem;
  }

  @media ${device.desktop} {
    font-size: ${({ theme }) => theme.fontSize.xl};
    padding: 0.6rem 0.8rem;
  }
`;
