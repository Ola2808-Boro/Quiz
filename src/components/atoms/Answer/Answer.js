import React from "react";
import PropTypes from "prop-types";
import { StyledAnswer } from "./Answer.styles";

const Answer = ({ isClicked, click, answer }) => {
  return (
    <StyledAnswer isClicked={isClicked} onClick={click}>
      {answer}
    </StyledAnswer>
  );
};

Answer.propTypes = {
  answer: PropTypes.string.isRequired,
  isClicked: PropTypes.bool.isRequired,
};

export default Answer;
