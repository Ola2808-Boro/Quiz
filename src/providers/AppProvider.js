import React from 'react';
import { GlobalStyle } from 'assets/styles/GlobalsStyle'
import { ThemeProvider } from 'styled-components';
import 'assets/styles/fonts.css'
import {theme} from 'assets/styles/Theme'


const AppProviders = ({ children }) => {
    return (
 
          <ThemeProvider theme={theme}>      
                <GlobalStyle />
                {children}
          </ThemeProvider>
 
    );
  };
  
  export default AppProviders;