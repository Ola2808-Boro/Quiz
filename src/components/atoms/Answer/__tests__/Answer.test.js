import "jest-styled-components";
import renderer from "react-test-renderer";
import AppProvider from "providers/AppProvider";
import React from "react";
import { StyledAnswer } from "components/atoms/Answer/Answer.styles";
import Answer from "components/atoms/Answer/Answer";
import { render, screen, rerender } from "@testing-library/react";
const mockfn = jest.fn();
describe("Test answer component", () => {
  it("default style", () => {
    const tree = renderer.create(
      <AppProvider>
        <StyledAnswer />
      </AppProvider>
    );
    expect(tree).toMatchSnapshot();
  });

  it("test style", () => {
    render(
      <AppProvider>
        <Answer isClicked={false} click={mockfn} answer="answer" />
      </AppProvider>
    );
    expect(screen.getByText(/answer/i)).toHaveStyle(
      `border: 1px solid "#d6dbf5"`
    );
    expect(screen.getByText(/answer/i)).toHaveStyle(
      `background-color: "transparent" `
    );
  });
  it("test style after click event", () => {
    render(
      <AppProvider>
        <Answer isClicked={true} click={mockfn} answer="answer" />
      </AppProvider>
    );
    expect(screen.getByText(/answer/i)).toHaveStyle(
      `border: 1px solid ${({ theme }) => theme.colors.borderAnswer}`
    );
    expect(screen.getByText(/answer/i)).toHaveStyle(
      `background-color: "#d6dbf5" `
    );
  });
});
