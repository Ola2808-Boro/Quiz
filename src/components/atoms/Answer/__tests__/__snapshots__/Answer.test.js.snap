// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test answer component default style 1`] = `
<div
  className="bKLFfl"
/>
`;
